lightspark-remote-signing/src/signing_requests.rs

```diff
@@ -0,0 +1,339 @@
+                ))
+            }
+            RemoteSigningSubEventType::ReleasePerCommitmentSecret => {
+                Ok(Self::ReleasePerCommitmentSecretRequest(
+                    ReleasePerCommitmentSecretRequest::parse_from_webhook_event(webhook_event)?,
+                ))
+            }
+            RemoteSigningSubEventType::DeriveKeyAndSign => Ok(Self::DeriveKeyAndSignRequest(
+                DeriveKeyAndSignRequest::parse_from_webhook_event(webhook_event)?,
+            )),
+            RemoteSigningSubEventType::RequestInvoicePaymentHash => {
+                Ok(Self::InvoicePaymentHashRequest(
+                    InvoicePaymentHashRequest::parse_from_webhook_event(webhook_event)?,
+                ))
+            }
+            RemoteSigningSubEventType::ReleasePaymentPreimage => {
+                Ok(Self::ReleasePaymentPreimageRequest(
+                    ReleasePaymentPreimageRequest::parse_from_webhook_event(webhook_event)?,
+                ))
+            }
+            RemoteSigningSubEventType::RevealCounterpartyPerCommitmentSecret => {
+                Ok(Self::ReleaseCounterpartyPerCommitmentSecretRequest(
+                    ReleaseCounterpartyPerCommitmentSecretRequest::parse_from_webhook_event(
+                        webhook_event,
+                    )?,
+                ))
+            }
+            _ => Err(Error::WebhookEventTypeNotSupported),
+        }
+    }
+}
+
+/// A signing request asking for a per-commitment point for a particular channel.
+/// The per-commitment point is the point on the secp256k1 curve for the commitment secret described
+/// in bolt 3.
+///
+/// [Bolt 3]: https://github.com/lightning/bolts/blob/master/03-transactions.md#per-commitment-secret-requirements
+#[derive(Clone, Deserialize)]
+pub struct GetPerCommitmentPointRequest {
+    pub channel_id: String,
+    pub derivation_path: String,
+    pub per_commitmnet_point_idx: u64,
+    pub bitcoin_network: BitcoinNetwork,
+}
+
+impl GetPerCommitmentPointRequest {
+    pub fn parse_from_webhook_event(webhook_event: &WebhookEvent) -> Result<Self, Error> {
+        let data = webhook_event
+            .data
+            .as_ref()
+            .ok_or(Error::WebhookEventDataMissing)?;
+        let data = data
+            .as_object()
+            .ok_or(Error::WebhookEventDataMissing)?
+            .clone();
+        let channel_id = webhook_event.entity_id.clone();
+        let derivation_path = data
+            .get("derivation_path")
+            .ok_or(Error::WebhookEventDataMissing)?
+            .as_str()
+            .ok_or(Error::WebhookEventDataMissing)?
+            .to_string();
+        let per_commitmnet_point_idx = data
+            .get("per_commitment_point_idx")
+            .ok_or(Error::WebhookEventDataMissing)?
+            .as_u64()
+            .ok_or(Error::WebhookEventDataMissing)?;
+        let bitcoin_network_str = data
```

Comment:
nit: It might be nice to break out a helper for parsing bitcoin_network from the data since you do this a lot.

Reply:
Yeah, will add in a follow-up PR.